### PR TITLE
test(google-adk): unit and integration tests

### PR DESCRIPTION
**Pull request by drPod**
## Summary
- Converter tests: span mapping, environment/customer propagation, session extraction, token counting, edge cases
- Instrumentor tests: patching, buffering, flush, deactivation
- Unit/e2e test suites extended with ADK instrumentation coverage

**Part 2 of 3** — depends on #101 (core package). Merge #101 first.

## Test plan
- [ ] `cd python-sdks/respan-instrumentation-google-adk && poetry run pytest`
- [ ] `cd python-sdks/examples && poetry run pytest test_unit.py -k "GoogleAdk"`

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: python-sdks/respan-instrumentation-google-adk/src/respan_instrumentation_google_adk/utils.py [hunk 1]
```diff
@@ -0,0 +1,418 @@
...
+        return None
+    if isinstance(value, str):
+        trimmed = value.strip()
+        if trimmed:
+            try:
+                json.loads(trimmed)
+                return trimmed
+            except Exception:
+                return json.dumps(value)
+        return json.dumps(value)
```
[the remark this answers is not in the document]
Comment (drPod) [reply python-sdks/respan-instrumentation-google-adk/src/respan_instrumentation_google_adk/utils.py:235]: This is the intended behavior — `serialize_value` produces valid JSON. A plain string `"hello"` serialized as JSON is `'"hello"'`. Downstream consumers that need plain text should use the raw value, not this function.

File: python-sdks/respan-instrumentation-google-adk/src/respan_instrumentation_google_adk/utils.py [hunk 2]
```diff
@@ -0,0 +1,418 @@
...
+        try:
+            return datetime.fromisoformat(trimmed.replace("Z", "+00:00"))
+        except ValueError:
+            try:
+                numeric_value = float(trimmed)
+                if reference and numeric_value < 1_000_000_000:
+                    return reference + timedelta(seconds=numeric_value)
+                if numeric_value > 100_000_000_000:
+                    return datetime.fromtimestamp(numeric_value / 1000, tz=timezone.utc)
+                return datetime.fromtimestamp(numeric_value, tz=timezone.utc)
```
[the remark this answers is not in the document]
Comment (drPod) [reply python-sdks/respan-instrumentation-google-adk/src/respan_instrumentation_google_adk/utils.py:174]: In the actual code flow, OTel nanosecond timestamps are converted to `datetime` objects by `ns_to_datetime()` in `otel_span_to_dict()` before reaching `coerce_datetime()`. So `coerce_datetime` never receives raw nanosecond values — it hits the `isinstance(value, datetime)` branch and returns directly.

File: python-sdks/respan-instrumentation-google-adk/tests/test_converter.py [hunk 3]
```diff
@@ -0,0 +1,757 @@
...
+        self.converter = AdkSpanConverter()
+
+    def test_invocation(self):
+        assert self.converter._map_log_type("invocation", None, None) == "workflow"
+
+    def test_agent_run(self):
+        assert self.converter._map_log_type("agent_run", "parent", None) == "agent"
+
+    def test_call_llm(self):
+        assert self.converter._map_log_type("call_llm", "parent", None) == "chat"
```
[the remark this answers is not in the document]
Comment (drPod) [reply python-sdks/respan-instrumentation-google-adk/tests/test_converter.py]: Good catch — fixing all `LOG_TYPE_CHAT` references to `LOG_TYPE_GENERATION` in the tests to match the converter mapping.

File: python-sdks/respan-instrumentation-google-adk/tests/test_converter.py [hunk 4]
```diff
@@ -0,0 +1,757 @@
...
+                    [{"role": "model", "content": "4"}]
+                ),
+            },
+        )
+        span_dict = otel_span_to_dict(span)
+        payloads = self.converter.convert(trace_or_spans=[span_dict])
+
+        assert len(payloads) == 1
+        payload = payloads[0]
+        assert payload["log_type"] == "chat"
```
[the remark this answers is not in the document]
Comment (drPod) [reply python-sdks/respan-instrumentation-google-adk/tests/test_converter.py]: Fixing — same issue as above.

File: python-sdks/respan-instrumentation-google-adk/pyproject.toml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+packages = [{include = "respan_instrumentation_google_adk", from = "src"}]
+
+[tool.poetry.dependencies]
+python = ">=3.12,<3.15"
+wrapt = "^1.16.0"
+opentelemetry-sdk = "^1.20.0"
+opentelemetry-api = "^1.20.0"
+opentelemetry-instrumentation = "^0.41b0"
+respan-sdk = ">=2.6.1"
+respan-ai = "*"
```
[the remark this answers is not in the document]
Comment (drPod) [reply python-sdks/respan-instrumentation-google-adk/pyproject.toml]: Fixing — pinning to `>= 3.0.0` to match `respan-instrumentation-openai-agents`.

File: python-sdks/respan-instrumentation-google-adk/tests/test_instrumentor.py [hunk 6]
```diff
@@ -0,0 +1,692 @@
...
+            llm.end()
+            agent.end()
+            inv.end()
+
+        assert len(captured_payloads) >= 9
+
+        log_types = [p["log_type"] for p in captured_payloads]
+        assert log_types.count(LOG_TYPE_WORKFLOW) >= 3
+        assert log_types.count(LOG_TYPE_AGENT) >= 3
+        assert log_types.count(LOG_TYPE_CHAT) >= 3
```
[the remark this answers is not in the document]
Comment (drPod) [reply python-sdks/respan-instrumentation-google-adk/tests/test_instrumentor.py]: Fixing all `LOG_TYPE_CHAT` references to `LOG_TYPE_GENERATION` across the test files.

File: python-sdks/respan-instrumentation-google-adk/tests/test_instrumentor.py [hunk 7]
```diff
@@ -0,0 +1,692 @@
...
+                "gen_ai.usage.input_tokens": 10 + i * 30,
+                "gen_ai.usage.output_tokens": 20,
+            }, context=trace_api.set_span_in_context(agent))
+            llm.end()
+            agent.end()
+            inv.end()
+
+        gens = [p for p in captured_payloads if p["log_type"] == LOG_TYPE_CHAT]
+        token_counts = sorted([g.get("prompt_tokens", 0) for g in gens])
+        assert token_counts == [10, 40, 70]
```
[the remark this answers is not in the document]
Comment (drPod) [reply python-sdks/respan-instrumentation-google-adk/tests/test_instrumentor.py]: Fixing — same batch of test updates.

File: python-sdks/respan-instrumentation-google-adk/tests/test_instrumentor.py [hunk 8]
```diff
@@ -0,0 +1,692 @@
...
+        agent.end()
+        inv.end()
+
+        assert len(captured_payloads) >= 3
+
+        gen = next(p for p in captured_payloads if p["log_type"] == LOG_TYPE_CHAT)
+        assert gen["prompt_tokens"] == 25
+        assert gen["completion_tokens"] == 200
+        assert gen["model"] == "gemini-2.0-flash"
+        assert "robot" in str(gen.get("output", "")).lower() or "pixel" in str(gen.get("output", "")).lower()
```
[the remark this answers is not in the document]
Comment (drPod) [reply python-sdks/respan-instrumentation-google-adk/tests/test_instrumentor.py]: Fixing.

File: python-sdks/respan-instrumentation-google-adk/tests/test_instrumentor.py [hunk 9]
```diff
@@ -0,0 +1,692 @@
...
+
+        type_counts = {}
+        for p in captured_payloads:
+            lt = p["log_type"]
+            type_counts[lt] = type_counts.get(lt, 0) + 1
+
+        assert type_counts.get(LOG_TYPE_WORKFLOW, 0) >= 1
+        assert type_counts.get(LOG_TYPE_AGENT, 0) >= 1
+        assert type_counts.get(LOG_TYPE_CHAT, 0) >= 2
+        assert type_counts.get(LOG_TYPE_TOOL, 0) >= 1
```
[the remark this answers is not in the document]
Comment (drPod) [reply python-sdks/respan-instrumentation-google-adk/tests/test_instrumentor.py:426]: Fixing.

File: python-sdks/respan-instrumentation-google-adk/tests/test_instrumentor.py [hunk 10]
```diff
@@ -0,0 +1,692 @@
...
+        self._emit_turn(tracer, 0, "Hi", "Hello")
+
+        wf = next(p for p in captured_payloads if p["log_type"] == LOG_TYPE_WORKFLOW)
+        agent = next(p for p in captured_payloads if p["log_type"] == LOG_TYPE_AGENT)
+        gen = next(p for p in captured_payloads if p["log_type"] == LOG_TYPE_CHAT)
+
+        assert wf["log_type"] == LOG_TYPE_WORKFLOW
+        assert agent["log_type"] == LOG_TYPE_AGENT
+        assert gen["log_type"] == LOG_TYPE_CHAT
+
```
[the remark this answers is not in the document]
Comment (drPod) [reply python-sdks/respan-instrumentation-google-adk/tests/test_instrumentor.py:582]: Fixing.

**Comment by drPod** [comment]
Closing: consolidated into PR #101. The exporter v2 approach is being replaced with the enricher-based architecture.
